gapis/gfxapi/vulkan/dependency_graph.go

```diff
@@ -0,0 +1,1733 @@
+				addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+				recordCommand(typedAtom.CommandBuffer, func(b *AtomBehaviour) {
+					writeMemoryBindingsData(b, bindingsToClear)
+				})
+			} else {
+				bindingsToRead := getOverlappedBindingsForImage(img)
+				addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+				addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+				recordCommand(typedAtom.CommandBuffer, func(b *AtomBehaviour) {
+					readMemoryBindingsData(b, bindingsToRead)
+				})
+			}
+		}
+
+	case *VkCmdEndRenderPass:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdEndRenderPass:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdNextSubpass:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdNextSubpass:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdPushConstants:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdPushConstants:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdSetLineWidth:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdSetLineWidth:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdSetScissor:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdSetScissor:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdSetViewport:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdSetViewport:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdBindDescriptorSets:
+		descriptorSetCount := typedAtom.DescriptorSetCount
```

Comment:
Probably needs further refactoring of the atom interface? Like creating a command buffer atom interface?